### PR TITLE
feat: Onboard perf recipes in tests

### PR DESCRIPTION
**Pull request by guyueh1**
# What does this PR do ?

Onboard some recipes for short perf testing in CI

# Issues
List issues that this PR closes ([syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)):

# Usage
* **You can potentially add a usage example below**

```python
# Add a code snippet demonstrating how to use this
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/NVIDIA-NeMo/RL/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you run the unit tests and functional tests locally? Visit our [Testing Guide](/NVIDIA-NeMo/RL/blob/main/docs/testing.md) for how to run tests
- [ ] Did you add or update any necessary documentation? Visit our [Document Development Guide](/NVIDIA-NeMo/RL/blob/main/docs/documentation.md) for how to write, build and test the docs.

# Additional Information
* ...

## Summary by CodeRabbit

* **New Features**
  * Added GRPO performance testing configurations for multiple large language models including DeepSeek v3, Llama 3.1/3.3, and Qwen3 series.
  * Introduced automated performance test scripts supporting single-node and multi-node distributed training setups.
  * Added integrated logging with WandB and TensorBoard, GPU monitoring, and metric validation capabilities.

### CODE REVIEW

File: examples/configs/recipes/llm/performance/grpo-deepseek-v3-32n8g.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  model_name: unsloth/DeepSeek-V3-0324-BF16
+  tokenizer:
+    name: unsloth/DeepSeek-V3-0324-BF16
+  train_micro_batch_size: 1
+  logprob_batch_size: 1
+  max_total_sequence_length: 1536
+  make_sequence_length_divisible_by: 1
+  dtensor_cfg:
+    enabled: false
+  megatron_cfg:
```
Comment (youngeunkwon0405) [inline examples/configs/recipes/llm/performance/grpo-deepseek-v3-32n8g.yaml:22]: @guyueh1, did you decide to remove TP=8? So, DSV3 is able to fit only with EP16PP16?

File: examples/configs/recipes/llm/performance/grpo-deepseek-v3-32n8g.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
+defaults: ../../../grpo_math_1B.yaml
+grpo:
+  num_prompts_per_step: 64
```
Comment (youngeunkwon0405) [inline examples/configs/recipes/llm/performance/grpo-deepseek-v3-32n8g.yaml]: Regarding the Rollout batch size, it is different from the spreadsheet. Could you please let me know what to follow?

Comment (guyueh1) [reply examples/configs/recipes/llm/performance/grpo-deepseek-v3-32n8g.yaml]: I will update this, should be 
- num_generations_per_prompt:16
- num_prompts_per_step:32

File: examples/configs/recipes/llm/performance/grpo-deepseek-v3-32n8g.yaml (hunk 1 above)
Comment (guyueh1) [reply examples/configs/recipes/llm/performance/grpo-deepseek-v3-32n8g.yaml:22]: No this recipe is waiting for testing, will update

**Comment by guyueh1** [comment]
@terrykong L0 CI is passing, should we run more CI?

**Review by terrykong — APPROVED** [review]
